lang/stack/Portfile

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github                      1.0
+
+github.setup        commercialhaskell stack 2.1.1
+
+name                stack
+categories          lang haskell
+maintainers         nomaintainer
+license             BSD
+platforms           darwin
+
+description The Haskell Tool Stack
+long_description    ${description}. \
+    Stack is a cross-platform program for developing Haskell projects. \
+    It is intended for Haskellers both new and experienced.
+
+homepage            http://haskellstack.org
+
+variant bootstrap \
+	    description {Bootstrap a build of stack.} {}
+
+if {[variant_isset "bootstrap"]} {
+    pre-fetch {
+        master_sites    ${github.homepage}/archive/v${github.version}
+        checksums       rmd160  60f7fcc5548aabec357be217d11d0d1d616c1806 \
+                        sha256  10c0119456748b189724ee42dea093c324d101487c3d75d57eb625675bb57424 \
+                        size    816217
+    }
+} else {
+    fetch {}
+    checksum {}
+    extract.cmd /usr/bin/true
```

Comment:
I don't quite get this bit of logic. You might want to comment it. (Indeed, you might want to comment the whole bootstrap thing, especially if you're not maintaining this, so others can understand it.)

Reply:
Now new logic based on your comments below and aversion to `curl | sh`.

The logic now is:
1. Download the pre-built binary from the github release page, then checksum.
2. Destroot the pre-built binary.
3. If bootstrapping, overwrite the pre-built binary.